src/Ext/BuildingType/Body.cpp

```diff
@@ -2,10 +2,48 @@
 
 #include <Ext/House/Body.h>
 #include <Utilities/GeneralUtils.h>
+#include <Ext/SWType/Body.h>
 
 template<> const DWORD Extension<BuildingTypeClass>::Canary = 0x11111111;
 BuildingTypeExt::ExtContainer BuildingTypeExt::ExtMap;
 
+int BuildingTypeExt::ExtData::GetSuperWeaponCount() const
+{
+	return 2 + this->SuperWeapons.Count;
+}
+
+int BuildingTypeExt::ExtData::GetSuperWeaponIndex(const int index, HouseClass* pHouse) const
+{
+	auto idxSW = this->GetSuperWeaponIndex(index);
+
+	if (auto pSuper = pHouse->Supers.GetItemOrDefault(idxSW))
+	{
+		auto pExt = SWTypeExt::ExtMap.Find(pSuper->Type);
+		if (!pExt->IsAvailable(pHouse))
+		{
+			return -1;
+		}
+	}
+
+	return idxSW;
+}
+
+int BuildingTypeExt::ExtData::GetSuperWeaponIndex(const int index) const
+{
+	const auto pThis = this->OwnerObject();
+
+	if (index < 2)
+	{
+		return !index ? pThis->SuperWeapon : pThis->SuperWeapon2;
+	}
+	else if (index - 2 < this->SuperWeapons.Count)
+	{
+		return this->SuperWeapons[index - 2]->ArrayIndex;
+	}
```

Comment:
```suggestion
	if (index < 2)
		return !index ? pThis->SuperWeapon : pThis->SuperWeapon2;
	else if (index - 2 < this->SuperWeapons.Count)
		return this->SuperWeapons[index - 2]->ArrayIndex;
```

Also magic numbers/underexplained on why 2, but I at least get that this is first 2 superweapons.